WordPress/gutenberg
# Document settings: Move 'Mark as pending', 'Revision history' and 'Trash post' to new ellipsis menu in Summary panel

## What?
- Makes the _Summary_ panel non-collapsible.
- Adds a ⋮ toolbar button that opens a menu.
- Moves _Stick to the top_, _Mark as pending_, _Revision history_ and _Trash post_ buttons into said menu.

## Why?
Part of https://github.com/WordPress/gutenberg/issues/39082.

## How?
All of the new functionality is contained within a new `PostStatusHeader` component in `@wordpress/edit-post`. This replaces the `PostSticky`, `PostPendingStatus`, `PostLastRevision` and `PostTrash` components in `@wordpress/edit-post`.

The `PostSticky`, `PostPendingStatus`, `PostLastRevision` and `PostTrash` components  components in `@wordpress/editor` remain as is. We can't touch these due to backwards compatibility. It's not great that this is now all dead code. I'm unsure whether we should deprecate them, try to use them in `PostStatusHeader` (difficult because we want menu items, not buttons), or do nothing. Open to suggestions.

I am thinking maybe `PostStatusHeader` should go into `@wordpress/editor` too so that non-WordPress post editors can make use of it. Not 100% sure of this yet.

## Testing Instructions
1. Create or edit a post.
2. Play with the various buttons
    - Stick to the top: Should always appear
    - Mark as pending: Appears when post was authored by a user with 'contributor' role
    - Revision history: Appears when post has been published more than once
    - Trash post: Appears when post is published 

## Screenshots or screencast 
https://user-images.githubusercontent.com/612155/177481576-248e5458-a5de-4a4b-a6c9-37df355f9177.mp4
